main.cpp

```diff
@@ -109,12 +109,14 @@ struct FloatType
     FloatType& apply( std::function< FloatType&( float& )> func )
     {
         if (func != nullptr )
-            func( *a );
-        return *this;
+            return func( *a );
+        else
+            std::cout << "Warning, nullptr, can't apply.\n";
+            return *this;
```

Comment:
You didn't use `{ }` so this `return *this` is not actually a part of that `else` expression.
- [ ] fix your indents.